src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -4682,6 +4739,19 @@ static TypeWithState calculateResultType(TypeWithAnnotations targetTypeWithNulla
             return operandType;
         }
 
+        private void SnapshotWalkerThroughConversionGroup(BoundConversion conversionOpt, BoundExpression convertedNode)
```

Comment:
>BoundConversion conversionOpt [](start = 58, length = 29)

All callers pass `expr as BoundConversion`. Consider changing this argument to `BoundExpression` and checking for `BoundConversion` here. (We're already performing that check inside the `while`.)